migration/migration.go

```diff
@@ -16,14 +17,23 @@ import (
 	"github.com/juju/version"
 	"gopkg.in/juju/charm.v6"
 
+	"github.com/juju/juju/core/leadership"
 	"github.com/juju/juju/core/migration"
+	"github.com/juju/juju/feature"
 	"github.com/juju/juju/resource"
 	"github.com/juju/juju/state"
 	"github.com/juju/juju/tools"
 )
 
 var logger = loggo.GetLogger("juju.migration")
 
+// When we import a new model, we need to give the leaders some time to
+// settle. We don't want to have leader switches just because we migrated an
+// model, so this time needs to be long enough to make sure we cover
+// the time taken to migration a reasonable sized model. We don't yet
+// know how long this is going to be, but we need something.
+const initialLeaderClaimTime = time.Minute
```

Comment:
Arbitrary times like this are a bit kudgy. I can see it could take longer than a minute waiting for all the agents to restart again etc. I wonder if we need a way to lock or freeze leadership claims and then unfreeze after migration.
Can we at least make it a todo to come back and take another look